torch_xla/_internal/gpu.py

```diff
@@ -26,13 +28,12 @@ def initialize_distributed_runtime(global_world_size: int) -> None:
     global_world_size: number of devices in the cluster.
   """
   if global_world_size > 1:
-    # TODO(jonbolin): For multi-host, this needs to be consistent across hosts
-    os.environ.setdefault(xenv.PJRT_DIST_SERVICE_ADDR, '127.0.0.1:8547')
     global distributed_service
     if distributed_service is None:
       num_nodes = global_world_size
       distributed_service = torch_xla._XLAC._xla_get_distributed_runtime_service(
           num_nodes)
+      atexit.register(shutdown_distributed_runtime)
```

Comment:
Neat! So this will shutdown the distributed runtime when the rank-0 process exits?

I'm curious if this process somehow finishes first, will the shutdown block until all clients are destroyed?

Reply:
TIL this module exists. Cool!

doc for other reviewers who are curious https://docs.python.org/3/library/atexit.html